### PR TITLE
Add cgroup v2 support

### PR DESCRIPTION
**Pull request by nbdd0121**
cgroup v1 uses two special files to determine access, where cgroup v2 uses eBPF programs to control access. The code will attach a custom eBPF program which allows run-time reconfiguration and detach docker's default.

eBPF programs will be detached when the attaching program dies, which can be dangerous if container-hotplug exits unexpectedly while the program is running, so we instead pin it (so it stays when the program exits) and unpin it after the docker container is down. In this case we might have garbage eBPF programs pinned when container-hotplug exits unexpectedly but it is safe.

### CODE REVIEW

**Comment by nbdd0121** [comment]
@jprendes just in case that you're interested

File: Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ bollard = "0.16"
 futures = "0.3"
 rustix = { version = "0.38", features = ["fs", "stdio", "termios"] }
 bitflags = "2"
+aya = { git = "https://github.com/nbdd0121/aya.git" }
```
Comment (pamaury) [inline Cargo.toml:33]: We should either merge changes upstream or fork the repo under lowRISC I think. Seems like your changes could be upstreamed?

Comment (jwnrt) [reply Cargo.toml:33]: It was actually [upstreamed one hour ago!](https://github.com/aya-rs/aya/commit/542306d295e51ac1ec117ce453544f201875af3d) Good timing :)

**Review by pamaury — APPROVED** [review]
This looks good to me, I have never really used eBPF programs so I might have missed something.

File: src/docker/cgroup.rs [hunk 2]
```diff
@@ -96,4 +104,95 @@ impl DeviceAccessController for DeviceAccessControllerV1 {
...
     }
+
+    fn stop(self: Box<Self>) -> Result<()> {
+        Ok(())
+    }
+}
+
+#[allow(unused)] // This is read as POD by the BPF program.
+#[derive(Clone, Copy)]
+struct Device {
```
[the remark this answers is not in the document]
Comment (nbdd0121) [reply src/docker/cgroup.rs:115]: This was `#[repr(C)]` but somehow I lost that during a rebase. Well spotted!

**Pull request merged**